templates/cli/lib/commands/run.js.twig

```diff
@@ -0,0 +1,283 @@
+const Tail = require('tail').Tail;
+const EventEmitter = require('node:events');
+const ignore = require("ignore");
+const tar = require("tar");
+const fs = require("fs");
+const ID = require("../id");
+const childProcess = require('child_process');
+const chokidar = require('chokidar');
+const inquirer = require("inquirer");
+const path = require("path");
+const { Command } = require("commander");
+const { localConfig, globalConfig } = require("../config");
+const { paginate } = require('../paginate');
+const { functionsListVariables } = require('./functions');
+const { usersGet, usersCreateJWT } = require('./users');
+const { projectsCreateJWT } = require('./projects');
+const { questionsRunFunctions } = require("../questions");
+const { actionRunner, success, log, error, commandDescriptions, drawTable } = require("../parser");
+const { systemHasCommand, isPortTaken, getAllFiles } = require('../utils');
+const { openRuntimesVersion, runtimeNames, systemTools, JwtManager, Queue } = require('../emulation/utils');
+const { dockerStop, dockerCleanup, dockerStart, dockerBuild, dockerPull, dockerStopActive } = require('../emulation/docker');
+
+const runFunction = async ({ port, functionId, noVariables, noReload, userId } = {}) => {
+    // Selection
+    if(!functionId) {
+        const answers = await inquirer.prompt(questionsRunFunctions[0]);
+        functionId = answers.function;
+    }
+
+    const functions = localConfig.getFunctions();
+    const func = functions.find((f) => f.$id === functionId);
+    if (!func) {
+        throw new Error("Function '" + functionId + "' not found.")
+    }
+
+    const runtimeName = func.runtime.split("-").slice(0, -1).join("-");
+    const tool = systemTools[runtimeName];
+
+    // Configuration: Port
+    if(port) {
+        port = +port;
+    }
+
+    if(isNaN(port)) {
+        port = null;
+    }
+
+    if(port) {
+        const taken = await isPortTaken(port);
+
+        if(taken) {
+            error(`Port ${port} is already in use by another process.`);
+            return;
+        }
+    }
+
+    if(!port) {
+        let portFound = fale;
+        port = 3000;
+        while(port < 3100) {
```

Comment:
while very rare, why limit the search to 100 ports? Why not set it to something higher like 6000 ?

Reply:
No idea how long that might take. And as you said, very rare. I wouldn' worry too much until we get some complains.